lib/pages/settings/view/source_page.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:ui';
+
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/widgets/scaffold.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class SourcePage extends StatefulWidget {
+  static const String routeName = '/sources';
+
+  @override
+  _SourcePageState createState() => _SourcePageState();
+}
+
+class _SourcePageState extends State<SourcePage> {
+  bool wantsOfficialInfo;
+  bool wantsOrganizationInfo;
+  bool wantsStudentsInfo;
```

Comment:
Make this a Map<String, bool> where the key is the source key. This way you can remove a lot of the duplicate code in this file, and it's more scalable. And writing `sourceSelected['official']` is imo just as readable as `wantsOfficialInfo`.

Reply:
replaced this with a Map structure